tensordict/nn/common.py

```diff
@@ -517,6 +517,16 @@ def select_out_keys(self, *out_keys):
                 is_shared=False)
 
         """
+        if len(out_keys) == 1:
+            if out_keys[0] not in self.out_keys:
+                err_msg = f"Can't select non existent key: {out_keys[0]}. "
+                try:
+                    if out_keys[0][0] in self.out_keys:
+                        err_msg += f"Are you passing the keys in a list? Try unpacking as: `{', '.join(out_keys[0])}`"
+                except Exception:
+                    pass
```

Comment:
what does this do?

Reply:
if you pass a list with just one element it checks if that element is a key. otherwise it prints a meaningful message suggesting a possible solution.
that is do not call mod.select_out_keys(["a","b","c"]) but mod.select_out_keys("a","b","c")

Reply:
Sure! I was wondering why the nested try/except. Can we get rid of the inner one? Why should it fail?
I'm just afraid that if something else is wrong (eg self.out_keys is void or smth strange like that) we display the wrong error message because every exception is skipped.

Reply:
I have done this just to be concise.
do you think something like `if isinstance(out_keys[0], (tuple,list)) and out_keys[0] and out_keys[0][0] in self.out_keys:` would be better?

Reply:
this is not superb I must admit
we maybe we can put the raise directly after exept and duplicate it after without the `from`

Reply:
we are not interested in other exceptions. it's just to provide an hint on the solution. switched to the if version